src/pages/index.js

```diff
@@ -17,131 +21,185 @@ import ModalWithForm from "../components/ModalWithForm.js";
+            .addLike(id)
+            .then((res) => {
+              card.updateLikes(res.likes);
+            })
+            .catch((err) =>
+              console.log(`An error occurred when adding a like: ${err}`)
+            );
+        }
+      },
+      cardSelector: selectors.cardTemplate
+    },
+    
+  );
+  return card.getView();
+};
 
 // Instance of the UserInfo class
-const userInfo = new UserInfo({
-  userNameSelector: ".profile__name",
-  userDescriptionSelector: ".profile__description",
-  userImageSelector: ".profile__image",
+const newUserInfo = new UserInfo({
+  nameSelector: "#profileName",
+  descSelector: "#profileDescription",
+  avatarSelector: "#profileImage",
 });
 
-// Create and enable form validators for editing profile and adding cards
-const editProfileValidator = new FormValidator(config, profileForm);
-editProfileValidator.enableValidation();
-const editCardValidator = new FormValidator(config, cardAddModal);
-editCardValidator.enableValidation();
+// get web server info, cards and user data.
+api
+  .getWebpageInfo()
+  .then(([cardData, userData]) => {
+    userId = userData._id;
+    cardSection = new Section(
+      {
+        items: cardData,
+        renderer: (data) => {
+          const cardElement = card(data);
+          cardSection.addItem(cardElement);
+        },
+      },
+      ".cards"
+    );
+    cardSection.renderItems();
+    newUserInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    });
+    newUserInfo.setAvatar(userData.avatar);
+  })
+  .catch((err) =>
+    console.log(
+      `An error occurred when loading initial user and card data: ${err}`
+    )
+  );
+
+const cardAddForm = new ModalWithForm("#cardAddModal", (data) => {
+  const newCard = { name: data['card-title-input'], link: data['card-image-input'] };
+  cardAddForm.setApiCalling(true);
+  api
+    .addCard(newCard)
+    .then((result) => {
+      const cardElement = card(result);
+      cardSection.addItem(cardElement);
+      cardAddForm.close();
+    })
+    .catch((err) =>
+      console.log(`An error occurred when loading new card data: ${err}`)
+    )
+    .finally(() => cardAddForm.setApiCalling(false));
+});
+cardAddForm.setEventListeners();
 
-// Create an instance of the Section class
-const section = new Section(
-  { items: initialCards, renderer: (item) => renderCard(item, section) },
-  ".cards"
+addCardButton.addEventListener("click", () => {
+  addFormValidator._toggleButtonState();
+  cardAddForm.open();
+});
+const addFormValidator = new FormValidator(
+  validationConfig,
+  selectors.addForm
 );
+addFormValidator.enableValidation();
+
+const editProfileForm = new ModalWithForm("#profileEditModal", (data) => {
+  editProfileForm.setApiCalling(true);
+  api.getProfileInfo(data)
+    .then(() => {
+      newUserInfo.setUserInfo(data);
+      editProfileForm.close();
+    })
+    .catch((err) =>
+      console.log(`An error occurred when loading user profile data: ${err}`)
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors